### PR TITLE
feat(rdb save): serializer flush to sink only on flush call

### PR DESCRIPTION
**Pull request by adiholden**
Signed-off-by: adi_holden <adi@dragonflydb.io>

### CODE REVIEW

File: src/server/rdb_save.h [hunk 1]
```diff
@@ -163,17 +158,17 @@ class RdbSerializer {
     return SaveString(std::string_view{reinterpret_cast<const char*>(buf), len});
   }
 
+  // TODO(Adi) : add flag to flush compressed blob to sink, move zstd serializer under RdbSerializer
```
Comment (adiholden) [inline src/server/rdb_save.h:161]: @romange I will do this in a different PR to keep PR small

File: src/server/snapshot.cc [hunk 2]
```diff
@@ -185,7 +185,7 @@ void SliceSnapshot::SerializeSingleEntry(DbIndex db_indx, const PrimeKey& pk, co
 
 bool SliceSnapshot::FlushSfile(bool force) {
   if (force) {
-    auto ec = rdb_serializer_->FlushMem();
+    auto ec = rdb_serializer_->FlushToSink(sfile_.get());
```
Comment (romange) [inline src/server/snapshot.cc:188]: I think `sfile_->val.size()` condition below will always be below 4096 so `FlushSfile(false)` will never flush and just aggregate the whole snapshot internally.

**Comment by dranikpg** [comment]
Oh damn, I've just refactored the whole snapshot 😆

**Comment by romange** [comment]
It's a small change, we can hold off with it.

File: src/server/snapshot.cc (hunk 2 above)
Comment (adiholden) [reply src/server/snapshot.cc:188]: you are so right

File: src/server/rdb_save.cc [hunk 3]
```diff
@@ -591,49 +591,37 @@ error_code RdbSerializer::SaveStreamConsumers(streamCG* cg) {
...
 
   // interrupt point.
-  RETURN_ON_ERR(sink_->Write(mem_buf_.InputBuffer()));
+  RETURN_ON_ERR(s->Write(mem_buf_.InputBuffer()));
   mem_buf_.ConsumeInput(sz);
 
   return error_code{};
 }
 
+size_t RdbSerializer::SerializedLen() {
```
Comment (romange) [inline src/server/rdb_save.cc:621]: make it const

**Review by romange — APPROVED** [review]
LGTM

**Pull request merged**